i3lock.c

```diff
@@ -1810,6 +1818,30 @@ int main(int argc, char *argv[]) {
                     fprintf(stderr, "ring-width must be a positive float; ignoring...\n");
                     ring_width = 7.0;
                 }
+                break;
+            case 404:
+                arg = optarg;
+                if (sscanf(arg, "%d", &polygon_sides) != 1)
+                    errx(1, "polygon-sides must be a number\n");
+                if (polygon_sides < 3 && polygon_sides != 0) {
+                    fprintf(stderr, "polygon-sides must be greater then 2 or 0; ignoring...\n");
+                    polygon_sides = 0;
```

Comment:
Yeah just short circuit here and error out (see other options here that do that for reference) instead of setting to 0.